src/cunumeric/matrix/contract_omp.cc

```diff
@@ -206,36 +297,59 @@ struct ContractImplBody<VariantKind::OMP, LegateTypeCode::COMPLEX128_LT> {
+    std::array<tblis::len_type, LEGION_MAX_DIM> lhs_shape_tblis;
+    std::array<tblis::stride_type, LEGION_MAX_DIM> lhs_strides_tblis;
+    convertToTBLIS<int64_t, tblis::len_type>(lhs_ndim, lhs_shape, lhs_shape_tblis.data());
+    convertToTBLIS<int64_t, tblis::stride_type>(lhs_ndim, lhs_strides, lhs_shape_tblis.data());
+    tblis_init_tensor_z(&lhs,
+                        lhs_ndim,
+                        lhs_shape_tblis.data(),
+                        reinterpret_cast<std::complex<double>*>(lhs_data),
+                        lhs_strides_tblis.data());
 
     tblis_tensor rhs1;
+    std::array<tblis::len_type, LEGION_MAX_DIM> rhs1_shape_tblis;
+    std::array<tblis::stride_type, LEGION_MAX_DIM> rhs1_strides_tblis;
+    convertToTBLIS<int64_t, tblis::len_type>(rhs1_ndim, rhs1_shape, rhs1_shape_tblis.data());
+    convertToTBLIS<int64_t, tblis::stride_type>(rhs1_ndim, rhs1_strides, rhs1_shape_tblis.data());
     tblis_init_tensor_z(
       &rhs1,
       rhs1_ndim,
-      rhs1_shape,
+      rhs1_shape_tblis.data(),
       reinterpret_cast<std::complex<double>*>(const_cast<complex<double>*>(rhs1_data)),
-      rhs1_strides);
+      rhs1_strides_tblis.data());
 
     tblis_tensor rhs2;
+    std::array<tblis::len_type, LEGION_MAX_DIM> rhs2_shape_tblis;
+    std::array<tblis::stride_type, LEGION_MAX_DIM> rhs2_strides_tblis;
+    convertToTBLIS<int64_t, tblis::len_type>(rhs2_ndim, rhs2_shape, rhs2_shape_tblis.data());
+    convertToTBLIS<int64_t, tblis::stride_type>(rhs2_ndim, rhs2_strides, rhs2_shape_tblis.data());
     tblis_init_tensor_z(
       &rhs2,
       rhs2_ndim,
-      rhs2_shape,
+      rhs2_shape_tblis.data(),
       reinterpret_cast<std::complex<double>*>(const_cast<complex<double>*>(rhs2_data)),
-      rhs2_strides);
+      rhs2_strides_tblis.data());
 
-    tblis_tensor_mult(nullptr, nullptr, &rhs1, rhs1_modes, &rhs2, rhs2_modes, &lhs, lhs_modes);
-  }
-};
+    std::array<tblis::label_type, LEGION_MAX_DIM> rhs1_modes_tblis, rhs2_modes_tblis,
+      lhs_modes_tblis;
+    convertToTBLIS<int32_t, tblis::label_type>(rhs1_ndim, rhs1_modes, rhs1_modes_tblis.data());
+    convertToTBLIS<int32_t, tblis::label_type>(rhs2_ndim, rhs2_modes, rhs2_modes_tblis.data());
+    convertToTBLIS<int32_t, tblis::label_type>(lhs_ndim, lhs_modes, lhs_modes_tblis.data());
+    tblis_tensor_mult(nullptr,
+                      nullptr,
+                      &rhs1,
+                      rhs1_modes_tblis.data(),
+                      &rhs2,
+                      rhs2_modes_tblis.data(),
+                      &lhs,
+                      lhs_modes_tblis.data());
+  };
 
-/*static*/ void ContractTask::omp_variant(legate::TaskContext& context)
-{
-  std::stringstream ss;
-  ss << omp_get_max_threads();
-  std::string str = ss.str();
-  setenv("TBLIS_NUM_THREADS", str.data(), false /*overwrite*/);
-  contract_template<VariantKind::OMP>(context);
-}
+  /*static*/ void ContractTask::omp_variant(legate::TaskContext& context)
```

Comment:
see [this comment](https://github.com/nv-legate/cunumeric/pull/384#issuecomment-1142699114). there's a reason that we don't use `tblis_set_num_threads`. I won't accept this change, unless we know for sure that the issue in that comment has been addressed.

Reply:
Huh, I use this in DISTAL and have never ran into that problem. I'll remove it from this PR though.